File: result/translate/chatgpt/java2python/python/MAXIMUM_DISTANCE_TWO_OCCURRENCES_ELEMENT_ARRAY.py
def f_gold(arr, n):
    map = {}
    max_dist = 0
    for i in range(n):
        if arr[i] not in map:
            map[arr[i]] = i
        else:
            max_dist = max(max_dist, i - map[arr[i]])
    return max_dist

File: result/translate/starchat/python2java/java/MAXIMUM_DISTANCE_TWO_OCCURRENCES_ELEMENT_ARRAY.java
public static int f_gold ( int arr [ ], int n ) {
        Map < Integer, Integer > mp = new HashMap < > ( );
        int maxDict = 0;
        for ( int i = 0; i < n; i++ ) {
            if (!mp.containsKey ( arr [ i ] ) ) {
                mp.put ( arr [ i ], i );
            } else {
                maxDict = Math.max ( maxDict, i - mp.get ( arr [ i ] ) );
            }
        }
        return maxDict;
    }